### PR TITLE
Add more information and examples to linear algebra manual

### PR DESCRIPTION
**Pull request by kshyatt**
* Move things around so factorizations come after special matrices
* Show some basic examples of calling linear algebra functions
* Discuss special matrix types besides just the table

### CODE REVIEW

File: doc/src/manual/linear-algebra.md [hunk 1]
```diff
@@ -1,29 +1,133 @@
...
+3×3 Array{Float64,2}:
+   1.5   2.0  -4.0
+   3.0  -1.0  -6.0
+ -10.0   2.3   4.0
+
+julia> factorize(A)
+Base.LinAlg.LU{Float64,Array{Float64,2}}([-10.0 2.3 4.0; -0.15 2.345 -3.4; -0.3 -0.132196 -5.24947],[3,3,3],0)
+```
+
+Since `A` is not Hermitian, symmertric, triangular, tridiagonal, or bidiagonal, an LU factorization may be the
```
Comment (tkelman) [inline doc/src/manual/linear-algebra.md]: symmetric

Comment (kshyatt) [inline doc/src/manual/linear-algebra.md]: Thanks :). I've only had one ☕️  today...

File: doc/src/manual/linear-algebra.md [hunk 2]
```diff
@@ -1,29 +1,133 @@
...
+
+julia> sB\x
+3-element Array{Float64,1}:
+ -1.73913
+ -1.1087
+ -1.45652
+```
+The `\\` operation here performs the linear solution. Julia's parser provides convenient dispatch
+to specialized methods for the *transpose* of a matrix left-divided by a vector, or for the various combinations
+of tranpose operations in matrix-matrix solutions. Many of these are further specialized for certain special
```
Comment (tkelman) [inline doc/src/manual/linear-algebra.md]: transpose

it's not immediately clear from the example above why transposed left-divide is relevant here

Comment (kshyatt) [inline doc/src/manual/linear-algebra.md]: Should I add another example with `@which` and a non-symmetric array?

Comment (tkelman) [inline doc/src/manual/linear-algebra.md]: maybe? did you mean the example to be `sB'\x` ?

Comment (kshyatt) [inline doc/src/manual/linear-algebra.md]: I want to show that either `B.'\x` or `sB.'\x` calls `At_ldiv_B`, I suppose, but that for the second one this is the same as `A_ldiv_B`.

Comment (kshyatt) [inline doc/src/manual/linear-algebra.md]: @tkelman Better/more clear now?

File: doc/src/manual/linear-algebra.md (hunk 1 above)
Comment (tkelman) [inline doc/src/manual/linear-algebra.md]: typo still present

File: doc/src/manual/linear-algebra.md (hunk 2 above)
Comment (tkelman) [inline doc/src/manual/linear-algebra.md]: the method naming of `A_ldiv_B` combines especially poorly with examples where the left hand operand is named `B` ... (the naming conventions of error messages in LinAlg also have this problem, they make sense given the internal argument naming but not with respect to user code)

Comment (kshyatt) [inline doc/src/manual/linear-algebra.md]: I can change my examples to use `A` and `B`?

**Comment by kshyatt** [comment]
@tkelman Better?

File: doc/src/manual/linear-algebra.md (hunk 2 above)
Comment (tkelman) [inline doc/src/manual/linear-algebra.md]: "tranpose" typo still present

Comment (kshyatt) [inline doc/src/manual/linear-algebra.md]: oh nice

**Review by tkelman — DISMISSED** [review]

**Comment by tkelman** [comment]
does this succeed at doing a doc build for you locally?

**Comment by kshyatt** [comment]
I'm fixing the `ldiv` refs now.

**Pull request merged**